external_executors/local_cromwell/run_portability_test.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+"""A script to run a WDL (and its subWDLs) in a local cromwell environment.
+
+This is placed within a docker container that has cromwell and tools for
+localizing inputs installed. It's run as the entrypoint to the container.
+
+There is one significant and very sad constraint on how this script can be
+run. This is being run in a docker container, and the WDL it executes can
+specify docker images in which to run tasks. So we've got a
+docker-within-docker situation. And, the docker socket available to this
+script and to cromwell is mounted from the calling environment, so you
+have to make sure any paths you try to mount are valid paths in the calling
+environment, not within this container.
+"""
+
+import argparse
+import json
+import os
+import subprocess
+import sys
+
+# This script is meant to be run within a docker container, and it's going to
+# have to assume that an external directory has been mounted somewhere. That
+# location is set here.
+IO_DIR = "/io"
+
+def localize_inputs(inputs_json):
+    """Some inputs to the WDL are files. We need to download them so they're local
+    to the instance of cromwell we're running.
+
+    Args:
+      inputs_json: Local path to the JSON file that would normally get passed to
+        cromwell with a '-i'.
+
+    Returns:
+      localized_inputs_dict: Dictionary where the keys in the original JSON that
+        referred to remote files have replaces with local paths.
+    """
+
+    inputs_dict = json.load(open(inputs_json))
+
+    localized_inputs_dict = {}
+
+    for entry, value in inputs_dict.items():
+        # TODO: There are at least three protocols to support here: http, s3
+        # and gs.
+        if value.startswith("https://") or value.startswith("http://"):
+            local_path = os.path.join(IO_DIR, os.path.basename(value))
+            subprocess.run(["curl", "-o", local_path, value])
```

Comment:
May what to have check = true so you can error handle here. No reason to attempt to run if you have an error at localizing files.

Reply:
Yeah good call.